src/js/components/modals/DeployServiceModal.js

```diff
@@ -0,0 +1,206 @@
+import mixin from 'reactjs-mixin';
+import {Modal} from 'reactjs-components';
+import React from 'react';
+import {StoreMixin} from 'mesosphere-shared-reactjs';
+
+import AdvancedConfig from '../AdvancedConfig';
+import InternalStorageMixin from '../../mixins/InternalStorageMixin';
+import MarathonStore from '../../stores/MarathonStore';
+import Service from '../../structs/Service';
+import ServiceUtil from '../../utils/ServiceUtil';
+import ServiceSchema from '../../constants/ServiceSchema';
+
+const METHODS_TO_BIND = [
+  'getAdvancedSubmit',
+  'onMarathonStoreServiceCreateSuccess',
+  'onMarathonStoreServiceCreateError',
+  'handleChange',
+  'handleSubmit'
+];
+
+class DeployServiceModal extends mixin(InternalStorageMixin, StoreMixin) {
+  constructor() {
+    super(...arguments);
+
+    this.internalStorage_set({
+      hasFormErrors: false,
+      pendingRequest: false
+    });
+
+    let model =
+      ServiceUtil.createFormModelFromSchema(new Service(), ServiceSchema);
```

Comment:
This should be able to also accept a `service` as a prop, which if not set is `new Service()` by default.